src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveResourceService.cs

```diff
@@ -84,36 +84,37 @@ public async Task<RetrieveResourceResponse> GetInstanceResourceAsync(RetrieveRes
 
         try
         {
-            string requestedTransferSyntax = _retrieveTransferSyntaxHandler
-                .GetTransferSyntax(message.ResourceType, message.AcceptHeaders, out AcceptHeaderDescriptor acceptHeaderDescriptor, out AcceptHeader acceptedHeader);
+            AcceptHeader validAcceptHeader = _retrieveTransferSyntaxHandler.GetValidAcceptHeader(
+                message.ResourceType,
+                message.AcceptHeaders);
 
-            bool isOriginalTransferSyntaxRequested = DicomTransferSyntaxUids.IsOriginalTransferSyntaxRequested(requestedTransferSyntax);
+            bool isOriginalTransferSyntaxRequested = DicomTransferSyntaxUids.IsOriginalTransferSyntaxRequested(validAcceptHeader.TransferSyntax.Value);
```

Comment:
It may make sense to set a variable still for `string requestedTransferSyntax = validAcceptHeader.TransferSyntax.Value;` since it seems we do this a number of times.